docs/source/deepspeed.mdx

```diff
@@ -0,0 +1,104 @@
+
+# Using DeepSpeed with Optimum Habana
+
+The [`GaudiTrainer`] allows to use DeepSpeed as easily as the [Transformers Trainer](https://huggingface.co/docs/transformers/main_classes/trainer).
+This can be done in 3 steps:
+1. A DeepSpeed configuration has to be defined.
+2. The `deepspeed` training argument enables to specify the path to the DeepSpeed configuration.
+3. The `deepspeed` launcher must be used to run your script.
+
+These steps are detailed below.
+A comprehensive guide about how to use DeepSpeed with the Transformers Trainer is also available [here](https://huggingface.co/docs/transformers/main_classes/deepspeed).
+
+
+## DeepSpeed configuration
+
+The DeepSpeed configuration to use is passed through a JSON file and enables you to choose the optimizations to apply.
+Here is an example for applying ZeRO-2 optimizations and *bf16* precision:
+```json
+{
+    "steps_per_print": 64,
+    "train_batch_size": "auto",
+    "train_micro_batch_size_per_gpu": "auto",
+    "gradient_accumulation_steps": "auto",
+    "bf16": {
+        "enabled": true
+    },
+    "gradient_clipping": 1.0,
+    "zero_optimization": {
+        "stage": 2,
+        "overlap_comm": false,
+        "reduce_scatter": false,
+        "contiguous_gradients": false
+    }
+}
+```
+
+<Tip>
+
+The special value `"auto"` enables to automatically get the correct or most efficient value.
+You can also specify the values yourself but, if you do so, you should be careful to not have conficting values with your training arguments.
+It is strongly advised to read [this section](https://huggingface.co/docs/transformers/main_classes/deepspeed#shared-configuration) in the Transformers documentation to completely understand how this works.
+
+</Tip>
+
+Other examples of configurations for HPUs are proposed [here](https://github.com/HabanaAI/Model-References/tree/1.6.0/PyTorch/nlp/pretraining/deepspeed-bert) by Habana.
+
+The [Transformers documentation](https://huggingface.co/docs/transformers/main_classes/deepspeed#configuration) explains how to write a configuration from scratch very well.
+A more complete description of all configuration possibilities is available [here](https://www.deepspeed.ai/docs/config-json/).
+
+
+## The `deepspeed` training argument
+
+In order to use DeepSpeed, you must specify `deespeed=path_to_my_deepspeed_configuration` in your `GaudiTrainingArguments` instance:
+```python
+training_args = GaudiTrainingArguments(
+    # my training arguments
+    use_habana=True,
+    use_lazy_mode=True,
+    gaudi_config_name=path_to_my_gaudi_config,
+    deepspeed=path_to_my_deepspeed_config,
+)
+```
+
+This argument both indicates that DeepSpeed should be used and points to your DeepSpeed configuration.
+
+
+## Launching your script
+
+Finally, you can run your script as follows:
+```bash
+deepspeed --num_nodes 1 --num_gpus 8 --no_local_rank path_to_my_training_script args
+```
```

Comment:
you could also use `torchrun` or `torch.distributed`. 

Also, should we add it to the `gaudi_spawn.py`?

Reply:
Great idea, I just added it in a new commit!